### PR TITLE
Elasticsearch persist data

### LINKED ISSUE #425: Logs are gone after I rebooted my laptop
First of all, docker-elk is awesome! I have all elasticsearch, logstash and kibana running on my mac osx. Yesterday, I had to reboot my laptop. I didn't delete the ELK containers. The command `docker ps -a` still shows the container. When I started it with `docker-compose up`, I noticed kibana and logstash was failing to connect to ElasticSearch. It was saying authorization failure. I had to run the command below again to generate a new password.
```
docker-compose exec -T elasticsearch bin/elasticsearch-setup-passwords auto --batch
```
After running the above, I updated logstash and kibana configuration files and ran
```
docker-compose restart kibana logstash
```
Kibana and Logstash was finally able to connect. However, it can't find the previous logs from yesterday. I tried last 24 hours and last 7 days but no luck.

In your doc it says that if container was deleted, that's the only time that data will be gone. However, I didn't delete the ELK containers. What might have happened? Is there something I'm missing?

### PR DESCRIPTION
**Pull request by deviantony**
Persist Elasticsearch data by default via a named volume for Compose deployment. 

Closes #425

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,8 +43,6 @@ Other available stack variants:
    * [How to configure Logstash](#how-to-configure-logstash)
    * [How to disable paid features](#how-to-disable-paid-features)
    * [How to scale out the Elasticsearch cluster](#how-to-scale-out-the-elasticsearch-cluster)
-4. [Storage](#storage)
-   * [How to persist Elasticsearch data](#how-to-persist-elasticsearch-data)
 5. [Extensibility](#extensibility)
```
Comment (antoineco) [inline README.md]:
```suggestion
4. [Extensibility](#extensibility)
```

File: README.md [hunk 2]
```diff
@@ -112,6 +110,16 @@ You can also run all services in the background (detached mode) by adding the `-
 
 If you are starting the stack for the very first time, please read the section below attentively.
 
+### Cleanup
```
Comment (antoineco) [inline README.md]: Should be added to the table of contents .

**Comment by deviantony** [comment]
@antoineco thanks, ready to merge after the checks.

**Comment by unstephenk** [comment]
I tried adding the following line to my docker-compose.yml and now I am getting a security node error.

```
- type: volume
        source: $PWD/elasticsearch/data
        target: /usr/share/elasticsearch/data
```

ERROR:
```
{"type": "server", "timestamp": "2020-02-27T17:36:20,019Z", "level": "ERROR", "component": "o.e.x.s.a.e.NativeUsersStore", "cluster.name": "docker-cluster", "node.name": "0844a39c5441", "message": "security index is unavailable. short circuiting retrieval of user [userXX]", "cluster.uuid": "TT6ipho7TR-RaXZ4xWs3sA", "node.id": "JLO2KhoDQLyrmDUX96W-qg"  }
```

**Comment by antoineco** [comment]
@unstephenk I don't think Compose accepts variables in volume definitions (`$PWD` will most likely be interpreted literally).

Please also read the note about user permissions carefully in the README.

**Comment by unstephenk** [comment]
Used the full filepath and read the permissions. still getting the same error.

**Comment by antoineco** [comment]
@unstephenk in that case please open a new issue and include _all_ the requested information (full change set, full logs, ...). This PR was closed 6 months ago already.
Thanks!

**Comment by unstephenk** [comment]
fixed it
`sudo chown -R $USER elasticsearch/data/`

**Pull request merged**